### PR TITLE
feature: add memberships model to describe related organizations

### PR DESCRIPTION
**Pull request by claire-lovisa**
## Issues

- Epic: OP-3195
- Separate issues: OP-3156, OP-3198, OP-3199, OP-3258, OP-3259, OP-3260,
  OP-3265, OP-3293, OP-3305

## Changes

- Introduce *memberships*, which have a *role*, as a new type of relationship
  between organizations.
- This new relationship replaces the relationships based on the
  `org:hasSubOrgzanization` and `ext:isFounderOfOrganization` predicates.
- Migrate the existing triples to the appropriate new ones.
- Update the necessary service configurations to handle the new kinds of
  resources and predicates.

## Frontend PR
Should be reviewed against https://github.com/lblod/frontend-organization-portal/pull/632

## TODO

- [X] Remove `participates-in` and `has-participants` relations by their
      equivalent `membership` relation.
- [x] Backup data in triplestore before deploying

### CODE REVIEW

File: config/resources/domain.json [hunk 1]
```diff
@@ -806,6 +767,68 @@
...
+          "cardinality": "one"
+        },
+        "has-broader-role": {
+          "predicate": "skos:broader",
+          "target": "membership-roles",
+          "cardinality": "one"
+        }
+      },
+      "new-resource-base": "http://data.lblod.info/id/rollen/"
+    },
```
Comment (aliokan) [inline config/resources/domain.json:817]: **Blocker:** `org:Role` is already defined https://github.com/lblod/app-organization-portal/blob/5ff3f5b1219f5c58c8cc686c76c32c3484da097b/config/resources/domain.json#L358-L366

File: config/resources/domain.json [hunk 2]
```diff
@@ -806,6 +767,68 @@
...
+        "top-concept-of": {
+          "predicate": "skos:topConceptOf",
+          "target": "concepts",
+          "cardinality": "one"
+        },
+        "has-broader-role": {
+          "predicate": "skos:broader",
+          "target": "membership-roles",
+          "cardinality": "one"
+        }
```
Comment (aliokan) [inline config/resources/domain.json:814]: **Question:** @bdevloed Do you know why `skos:broader` is used here?

**Review by aliokan — COMMENTED** [review]
@bdevloed We left some questions, Can you take a look? :pray:

File: config/migrations/2023/20230824160000-related-organizations/20230831104300-participants.sparql [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+INSERT {
+  GRAPH ?g {
+    ?membership
+      a org:Membership;
+      mu:uuid ?uuid;
+      org:organization ?organization;
+      org:member ?memberOrganization;
+      org:role <http://data.lblod.info/id/rollen/2152eb830b1143bfb97a7dd9596d6c63>.
+  }
+}
```
[the remark this answers is not in the document]
Comment (bdevloed) [reply config/migrations/2023/20230824160000-related-organizations/20230831104300-participants.sparql:14]: Erm, not really

File: config/resources/domain.json (hunk 2 above)
Comment (bdevloed) [reply config/resources/domain.json:900]: To build a kind of hierarchy. lid is the most general membership. And then there are some more special kinds of memberships.

File: config/resources/domain.json (hunk 1 above)
Comment (bdevloed) [reply config/resources/domain.json:903]: Use the skos concept scheme?

File: config/migrations/2023/20230824160000-related-organizations/20230831104400-related-members.sparql [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  GRAPH ?g {
+    ?subOrganization org:memberOf ?organization.
+
+    ?membership
+      a org:Membership;
+      mu:uuid ?uuid;
+      org:organization ?organization;
+      org:member ?subOrganization;
+      org:role <http://data.lblod.info/id/rollen/4ec7d5c39bdc4e84b4174379b9e22ad8>.
+  }
```
[the remark this answers is not in the document]
Comment (bdevloed) [reply config/migrations/2023/20230824160000-related-organizations/20230831104400-related-members.sparql:20]: cf above

**Review by bdevloed — APPROVED** [review]

**Pull request merged**